### ISSUE TITLE
[Feature] Interview Prep Checklist — Topic Tracker with Confidence Ratings

### ISSUE THREAD

**Issue by PascalBurume** [PascalBurume/ai-academy-demo-main#5]
## Summary

Build the `/jobs/interview-prep` page — a structured checklist that helps students systematically prepare for AI engineering interviews across four topic areas: ML Theory, System Design, Coding, and Behavioral. Students tick off questions as they practice them and rate their confidence, seeing progress toward interview readiness at a glance.

## Context & Motivation

Persona **Alex** knows he needs to prepare for technical interviews but doesn't know where to start or how to track his progress. Dedicated interview-prep platforms (LeetCode, Pramp) are unorganised and disconnected from his learning path. This checklist lives inside AI Academy and surfaces questions relevant to the AI roles he's targeting.

The key differentiator: this checklist ships with **seeded, AI-specific questions** (not generic LeetCode fare) and links preparation directly to the courses the student has completed on the platform.

Part of #1 — Epic: AI Jobs Application Module. Blocked by #2 — Schema Migration.

## Demo Scope

- **4 topic cards** shown as a grid: ML Theory, System Design, Coding, Behavioral
- **Each card** shows: topic name, question count, % practiced progress bar, and a "Start" / "Continue" button
- **Topic detail view** (expand in-page or separate route): full question list for that topic, each with:
  - Question text
  - "Practiced" checkbox toggle
  - Confidence selector (1 = shaky → 5 = confident)
  - Optional notes/answer field
- **Seeded questions**: ship with 6–8 starter questions per topic (PRE-SEEDED via a Prisma seed script) so the checklist is immediately usable on first load
- **Progress persistence**: checks and confidence ratings saved to DB per user
- **Overall readiness indicator**: a single "Interview Readiness" percentage across all topics shown in the page header

## User Stories Addressed

- **US-JOB-04** — Tracker organised by topic (ML Theory, System Design, Coding, Behavioral)
- **US-JOB-05** — Curated question bank by topic
- **US-JOB-06** — Log practice results, track confidence over time

### Page & Route
- [ ] Page exists at `/jobs/interview-prep`, protected for authenticated STUDENT users
- [ ] Unauthenticated → `/signin`; non-STUDENT → `/unauthorized`
- [ ] Uses `requireAuth()` and `checkRole([UserRole.STUDENT])` from `src/lib/auth/utils.ts`

### Seed Data
- [ ] Prisma seed script (`prisma/seed.ts`) creates `InterviewPrepTopic` rows and ≥ 6 starter questions per topic (24+ questions total)
- [ ] Seed is idempotent (safe to run multiple times with `upsert`)

**Starter questions by topic (minimum):**

*ML Theory*: Explain the bias-variance tradeoff · What is gradient descent and its variants? · How does a transformer work? · What is regularisation and when would you use it? · Explain precision vs. recall trade-off · What is cross-entropy loss?

*System Design*: Design a real-time recommendation system · How would you serve a large language model at scale? · Design a feature store for ML pipelines · How would you handle model drift in production? · Describe an MLOps pipeline from training to deployment · How do you design an A/B testing system for ML models?

*Coding*: Implement k-means clustering from scratch · Write a function to compute cosine similarity · Implement a simple neural network forward pass in Python · Given a dataset, identify and handle class imbalance · Reverse a linked list · Find the longest common subsequence

*Behavioral*: Tell me about a project where your model performed poorly — what did you do? · Describe a time you had to explain a complex ML concept to a non-technical stakeholder · How do you stay current with the fast pace of AI research? · Tell me about a time you disagreed with a team member on a technical decision · Describe your process for debugging a machine learning model

### API
- [ ] `GET /api/jobs/interview-prep/topics` returns all topics with per-user progress stats
- [ ] `POST /api/jobs/interview-prep/entries` creates or updates a `InterviewPrepEntry` for a given `(userId, topicId, question)` pair; validates `confidence` is an integer 1–5; validates `topicId` exists
- [ ] No raw `error.message` in responses

### UI
- [ ] 4 topic cards in a responsive grid (2-col desktop, 1-col mobile)
- [ ] Each card: topic name, category badge, progress bar (% questions practiced), question count
- [ ] "Interview Readiness" percentage header (avg across all topics)
- [ ] Topic detail: question list with checkbox, confidence selector (star rating or 1–5 buttons), and optional notes textarea
- [ ] Checking a question immediately persists via API (optimistic update)
- [ ] Empty state for new users (before any questions are practiced) with a motivational CTA
- [ ] Loading skeleton while topics fetch
- [ ] Uses `Button` from `src/components/ui/Button.tsx` and `cn()` from `src/lib/utils.ts`

### Quality
- [ ] `npm run build` passes with no TypeScript errors
- [ ] `npm run lint` passes with no new warnings
- [ ] Seed script runs cleanly: `npx ts-node prisma/seed.ts` (or `npx prisma db seed`)
- [ ] No new broken internal links

### File locations to create
```
src/app/jobs/interview-prep/page.tsx                ← Server component (thin shell + initial data)
src/app/jobs/interview-prep/PrepChecklist.tsx        ← Client component (interactive checklist)
src/app/api/jobs/interview-prep/topics/route.ts      ← GET topics + user progress
src/app/api/jobs/interview-prep/entries/route.ts     ← POST (upsert practice entry)
src/lib/jobs/repositories/interviewPrep.ts           ← All Prisma calls
src/lib/jobs/services/interviewPrep.ts               ← Business logic + progress calculation
prisma/seed.ts                                       ← Seed topics and starter questions
```

### Progress calculation
```typescript
// Per-topic readiness = (questions practiced / total questions) * 100
// Overall readiness = average of per-topic readiness scores
```

### Security
- Filter all queries by `userId` from the session token — never from request body
- `topicId` on entry creation must be validated to exist in `InterviewPrepTopic` table

## Related Issues

- Part of #1 — Epic: AI Jobs Application Module
- Blocked by #2 — Schema Migration
- Good first issue for the jobs module — zero AI dependency, clear data model, immediate user value
